libkbfs/folder_branch_ops.go

```diff
@@ -1891,10 +1892,15 @@ func newBlockPutState(length int) *blockPutState {
 // complete (whether or not the put resulted in an error).  Currently
 // it will not be called if the block is never put (due to an earlier
 // error).
-func (bps *blockPutState) addNewBlock(blockPtr BlockPointer, block Block,
+func (bps *blockPutState) addNewBlock(
+	blockPtr BlockPointer, block Block,
 	readyBlockData ReadyBlockData, syncedCb func() error) {
 	bps.blockStates = append(bps.blockStates,
-		blockState{blockPtr, block, readyBlockData, syncedCb})
+		blockState{blockPtr, block, readyBlockData, syncedCb, zeroPtr})
+}
+
+func (bps *blockPutState) saveOldPtr(oldPtr BlockPointer) {
+	bps.blockStates[len(bps.blockStates)-1].oldPtr = oldPtr
```

Comment:
Why is it always in the last `blockState`? Is it because in the future this will used *while* building out the `blockPutState.blockStates`?